src/NetMQ/Core/Patterns/XPub.cs

```diff
@@ -176,15 +184,29 @@ protected override void XReadActivated(Pipe pipe)
                         // If the subscription is not a duplicate, store it so that it can be
                         // passed to used on next recv call.
                         if (m_options.SocketType == ZmqSocketType.Xpub && (unique || m_verbose))
-                            m_pending.Enqueue(sub.CloneData());
+                        {
+                            m_pending.Enqueue(sub);
+                        }
+                        else
+                        {
+                            sub.Close();
+                        }
+
                     }
                 }
+                else if (m_broadcastEnabled && size > 0 && sub.HasMore && sub[0] == 2)
```

Comment:
because only the first part with the topic is prefixed. The second part, in general, could have 2 as the first byte.